nixos/modules/security/pam.nix

```diff
@@ -202,6 +204,61 @@ let
           '';
         };
 
+        pwquality = {
+          enable = lib.mkOption {
+            default = config.security.pam.pwquality.enable;
+            defaultText = lib.literalExpression "config.security.pam.pwquality.enable";
+            type = lib.types.bool;
+            description = ''
+              Enable PAM pwquality system for this service to enforce complex passwords.
+            '';
+          };
+
+          package = lib.mkOption {
+            default = config.security.pwquality.package;
+            defaultText = lib.literalExpression "config.security.pwquality.package";
+            type = lib.types.package;
+            description = ''
+              The libpwquality package to use.
+            '';
+          };
+
+          control = lib.mkOption {
+            default = config.security.pam.pwquality.control;
+            defaultText = lib.literalExpression "config.security.pam.pwquality.control";
+            type = lib.types.enum [
+              "required"
+              "requisite"
+              "sufficient"
+              "optional"
+            ];
+            description = ''
+              This option sets pam "control".
+              If you want to have multi factor authentication, use "required".
+              If you want to use U2F device instead of regular password, use "sufficient".
+
+              Read {manpage}`pam.conf(5)` for better understanding of this option.
+            '';
+          };
+
+          settings = lib.mkOption {
+            default = config.security.pam.pwquality.settings;
+            defaultText = lib.literalExpression "config.security.pam.pwquality.settings";
+            type = pwqualityConf.type;
+            apply = pwqualityConf.pamApply;
```

Comment:
nit: I missed this (`pamApply`) on my first pass through the new version. It would be nice to comment explaining what it does here. That way, if/when this option gets refactored out as part of bigger PAM refactors, this `apply` is less likely to be accidentally dropped.